### ISSUE TITLE
Pipeline APIs does not work with objects having identical name

### ISSUE THREAD

**Issue by saivirtue** [oharastream/ohara#1653]
**Describe the bug**
Since we use the `name` as each object unique id, same object could not have same name ; However, different object should able to use same name but UI will throw exception...

**To Reproduce**
1. Create a workspace, name **wk**
2. Create new topic in **wk**, name **abc**
3. Create new pipeline in **wk**, name **abc**
4. put a topic, select **abc**, and exception will thrown
```
abc exists on multiples types
```

full stack trace :
```
java.lang.IllegalStateException: abc exists on multiples types
	at com.island.ohara.configurator.store.DataStore.$anonfun$raw$1(DataStore.scala:71)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)↵
```

**Comment by saivirtue** [comment]
/cc @chia7712 @jackyoh 
this issue has minor effect for user, we could fix this in next version (0.7)

**Comment by chia7712** [comment]
Let me take over and check this later.
